File: packages/main-api/src/types/space_publish_state.rs
use bdk::prelude::*;

#[derive(
    Debug,
    Clone,
    Copy,
    serde_with::SerializeDisplay,
    serde_with::DeserializeFromStr,
    Default,
    DynamoEnum,
    JsonSchema,
    Eq,
    PartialEq,
)]
pub enum SpacePublishState {
    #[default]
    Draft,
    Published,
}

File: flutter-packages/mobile/lib/services/space/types/space_publish_state.dart
enum SpacePublishState { draft, published }

SpacePublishState spacePublishStateFromJson(dynamic v) {
  if (v == null) return SpacePublishState.draft;

  final s = v.toString().toLowerCase();
  switch (s) {
    case 'published':
      return SpacePublishState.published;
    case 'draft':
    default:
      return SpacePublishState.draft;
  }
}

String spacePublishStateToJson(SpacePublishState v) {
  switch (v) {
    case SpacePublishState.draft:
      return 'draft';
    case SpacePublishState.published:
      return 'published';
  }
}
